myia/compile/backends/relay.py

```diff
@@ -582,32 +605,66 @@ def ref(self, node):
 
     def convert_func(self, graph):
         """Convert a graph."""
+        vname = "handle_seq" + graph.debug.debug_name
+        i = 0
         for p in graph.parameters:
             self.node_map[p] = self.on_parameter(p)
 
         params = [self.ref(p) for p in graph.parameters]
 
+        useq = []
         for node in toposort(graph.output, NodeVisitor()):
-            if node.is_apply():
+            if any(p.abstract.xtype() is UniverseType
+                   for p in node.inputs[1:]):
+                useq.append(node)
+                self.node_map[node] = relay.var(f"{vname}.{i}")
+                i += 1
+            elif node.is_apply():
                 self.node_map[node] = self.on_apply(node)
             elif node.is_constant_graph():
                 self.node_map[node] = self.on_graph(node)
             elif node.is_constant():
                 self.node_map[node] = self.on_constant(node)
 
-        return relay.Function(params, self.ref(graph.output),
-                              ret_type=to_relay_type(graph.output.abstract))
+        out = self.ref(graph.output)
+
+        for uop in reversed(useq):
+            assert node.is_apply()
+            out = relay.Let(
+                self.node_map[uop],
+                self.on_apply(uop),
+                out)
+
+        res = relay.Function(params, out,
+                             ret_type=to_relay_type(graph.output.abstract))
+
+        return res
 
 
 compiler = CompileGraph()
 
 
+class DummyManager:
+    all_nodes = []
+
+
 class RelayInputConverter(Converter):
     """Convert values to Relay."""
 
     def __init__(self, context):
         """Set the context."""
         self.context = context
+        self.cst_conv = RelayConstantConverter(self.context)
+        mod = relay.Module({})
+        th = TypeHelper()
+        th.initialize(mod, DummyManager())
```

Comment:
Maybe edit TypeHelper.initialize so that the manager is optional?

Reply:
Done.